src/Tearing/TearingScenarioEngine.inl

```diff
@@ -0,0 +1,176 @@
+#pragma once
+#include <Tearing/TearingScenarioEngine.h>
+#include <sofa/simulation/AnimateBeginEvent.h>
+#include <sofa/simulation/AnimateEndEvent.h>
+#include <sofa/core/objectmodel/KeypressedEvent.h>
+
+#include <sofa/component/mechanicalload/ConstantForceField.h>
+#include <sofa/component/constraint/projective/FixedProjectiveConstraint.h>
+
+namespace sofa::component::engine
+{
+
+using sofa::type::Vec3;
+
+template <class DataTypes>
+TearingScenarioEngine<DataTypes>::TearingScenarioEngine()
+        : d_startVertexId(initData(&d_startVertexId, int(-1), "startVertexId", "Vertex ID to start a given tearing scenario, -1 if none"))
+        , d_startTriId(initData(&d_startTriId, int(-1), "startTriangleId", "Triangle ID from which the starting point is chosen, -1 if none"))
+        , d_startDirection(initData(&d_startDirection, Vec3(1.0, 0.0, 0.0), "startDirection", "If startVertexId is set, define the direction of the tearing scenario. x direction by default"))
+        , d_startLength(initData(&d_startLength, Real(1.0), "startLength", "If startVertexId is set, define the length of the tearing, to be combined with startDirection"))
+    {
+
+        
+        addInput(&d_startVertexId);
+        addInput(&d_startTriId);
+        addInput(&d_startDirection);
+        addInput(&d_startLength);
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::init()
+    {
+        BaseTearingEngine<DataTypes>::init();
+
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::reinit()
+    {
+        BaseTearingEngine<DataTypes>::update();
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::handleEvent(sofa::core::objectmodel::Event* event)
+    {
+        BaseTearingEngine<DataTypes>::handleEvent(event);
+    }
```

Comment:
This one should should override the BaseTearingEngine to only launch the algofracture Path once.